src/ImageSharp/Advanced/AdvancedImageExtensions.cs

```diff
@@ -40,23 +41,66 @@ public static Configuration GetConfiguration(this ImageFrame source)
             => GetConfiguration((IConfigurationProvider)source);
 
         /// <summary>
-        /// Gets the configuration .
+        /// Gets the configuration.
         /// </summary>
         /// <param name="source">The source image</param>
         /// <returns>Returns the bounds of the image</returns>
         private static Configuration GetConfiguration(IConfigurationProvider source)
             => source?.Configuration ?? Configuration.Default;
 
         /// <summary>
-        /// Gets the representation of the pixels as a <see cref="Span{T}"/> of contiguous memory in the source image's pixel format
-        /// stored in row major order.
+        /// Gets the representation of the pixels as a <see cref="IMemoryGroup{T}"/> containing the backing pixel data of the image
+        /// stored in row major order, as a list of contiguous <see cref="Memory{T}"/> blocks in the source image's pixel format.
         /// </summary>
+        /// <param name="source">The source image.</param>
         /// <typeparam name="TPixel">The type of the pixel.</typeparam>
-        /// <param name="source">The source.</param>
+        /// <returns>The <see cref="IMemoryGroup{T}"/>.</returns>
+        /// <remarks>
+        /// Certain Image Processors may invalidate the returned <see cref="IMemoryGroup{T}"/> and all it's buffers,
```

Comment:
Is this actually possible? All processors that alter the dimensions of an image operate on a new image with the expected output dimensions.

Reply:
Here is a test case that demonstrates the issue:
https://github.com/SixLabors/ImageSharp/blob/0675c80d2731396bdb964c4b5c587fee70e3b0a7/tests/ImageSharp.Tests/Advanced/AdvancedImageExtensionsTests.cs#L44-L53

Without `MemoryGroupView`, this is what would happen:

1. You take a reference to a memory area within the image with `Memory<TPixel>`
2. After Resize, the referenced backing buffer is returned to the pool during the Swap operations. 
3. The `Memory<T>` is referencing a buffer that has nothing to do with the image instance. Worst case: it is got rented by some other code leading to strange bugs, and safety issues.

This problem was originally pointed out by @tocsoft, and this is why we haven't exposed `GetPixelMemory()` and `GetPixelRowMemory()`.
`MemoryGroupView` is protecting against these bugs by adding a layer of indirection, so we can throw an exception when the user aims to access memory area that no longer belongs to the image.

Note that this can still happen by taking a row span, resizing an image and accessing the span later. But normally no one should hit this issue, since spans are intended to be short living objects for quick processing of data. `Memory<T>` is heapable, one can store a long-living reference in an class, or long-running async closure, without being aware that it gets outdated.

We should elaborate on this in our conceptual docs.

Reply:
Ah yeah, I totally understand the risk now. Yeah, lot's of warnings.